Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerTest.java

```diff
@@ -268,5 +263,75 @@ void WhenSaveProblemWithTitle_ProblemShouldBeSavedAndRespondItsId() throws Excep
 
         assertEquals(randomTitle, requireNonNull(mongoTemplate.findById(id, Problem.class)).getTitle());
     }
+
+    @Test
+    void GivenOneProblemSavedAndPatchProblemWithTitle_WhenQueryById_ShouldHaveNewTitle() throws Exception {
+        Problem savedProblem = givenProblemsSaved(1).get(0);
+        String newTitle = UUID.randomUUID().toString();
+        mockMvc.perform(patch("/api/problems/{problemId}/title", savedProblem.getId())
+                .contentType(MediaType.TEXT_PLAIN_VALUE).content(newTitle))
+                .andExpect(status().isOk());
+        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
+        assertNotNull(queryProblem);
+        assertEquals(newTitle, queryProblem.getTitle());
+    }
+
+    @Test
+    void GivenOneProblemSavedAndPatchProblemWithDescription_WhenQueryById_ShouldHaveNewDescription() throws Exception {
+        Problem savedProblem = givenProblemsSaved(1).get(0);
+        String newDescription = UUID.randomUUID().toString();
+        mockMvc.perform(patch("/api/problems/{problemId}/description", savedProblem.getId())
+                .contentType(MediaType.TEXT_PLAIN_VALUE).content(newDescription))
+                .andExpect(status().isOk());
+        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
+        assertNotNull(queryProblem);
+        assertEquals(newDescription, queryProblem.getDescription());
+    }
+
+    @Test
+    void GivenOneProblemSavedAndPatchProblemWithPluginMatchTags_WhenQueryById_ShouldHaveNewTags() throws Exception {
+        Problem savedProblem = givenProblemsSaved(1).get(0);
+        JudgePluginTag pluginMatchTag = new JudgePluginTag();
+        pluginMatchTag.setGroup("Judge Girl");
+        pluginMatchTag.setName("Test");
+        pluginMatchTag.setVersion("1.0");
+        pluginMatchTag.setType(JudgePluginTag.Type.OUTPUT_MATCH_POLICY);
+        mockMvc.perform(patch("/api/problems/{problemId}/plugins/match", savedProblem.getId())
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(objectMapper.writeValueAsString(pluginMatchTag)))
+                .andExpect(status().isOk());
+        Problem queryProblem = mongoTemplate.findById(savedProblem.getId(), Problem.class);
+        assertNotNull(queryProblem);
+        JudgePluginTag queryPluginMatchTag = queryProblem.getOutputMatchPolicyPluginTag();
+        assertEquals(
+                objectMapper.writeValueAsString(pluginMatchTag),
+                objectMapper.writeValueAsString(queryPluginMatchTag));
+    }
```

Comment:
Don't use JSON to check the equality as JSON is text-volatile and causes unstableness to our test.